infrastructure/code/snyk_code_http_client.go

```diff
@@ -134,50 +134,47 @@ func (s *SnykCodeHTTPClient) CreateBundle(
 		return "", nil, err
 	}
 
-	responseBody, err := s.doCall(span.Context(), "POST", "/bundle", requestBody)
+	body, err, _ := s.doCall(span.Context(), "POST", "/bundle", requestBody) //nolint:bodyclose // false positive
 	if err != nil {
 		return "", nil, err
 	}
 
 	var bundle bundleResponse
-	err = json.Unmarshal(responseBody, &bundle)
+	err = json.Unmarshal(body, &bundle)
 	if err != nil {
 		return "", nil, err
 	}
 	s.c.Logger().Debug().Str("method", method).Msg("API: Create done")
 	return bundle.BundleHash, bundle.MissingFiles, nil
 }
 
-func (s *SnykCodeHTTPClient) doCall(ctx context.Context,
-	method string,
-	path string,
-	requestBody []byte,
-) (responseBody []byte, _ error) {
+func (s *SnykCodeHTTPClient) doCall(ctx context.Context, method string, path string, requestBody []byte) ([]byte, error, *http.Response) {
```

Comment:
Questions about this, based on my limited understanding and lack of context:
- Why was the `*http.Response` return value added? I don't see any of the `doCall` callers using it.
- Where is this response's `.Body` getting closed? I'm concerned this is going to leak.

ISTM that the returned response should be used by all the callers to `defer resp.Body.Close()` to avoid the latter.

Reply:
1. response is used in line 330
2. body is always closed in doCall()